parts/k8s/cloud-init/jumpboxcustomdata.yml

```diff
@@ -11,12 +11,16 @@ write_files:
 
 - path: "/home/{{WrapAsParameter "jumpboxUsername"}}/.kube/config"
   permissions: "0644"
-  owner: "root"
+  owner: "{{WrapAsParameter "jumpboxUsername"}}"
   content: |
 {{WrapAsVariable "kubeconfig"}}
 
 runcmd:
 - . /opt/azure/containers/provision_source.sh
 - retrycmd_if_failure 10 5 10 curl -LO https://storage.googleapis.com/kubernetes-release/release/v{{.OrchestratorProfile.OrchestratorVersion}}/bin/linux/amd64/kubectl
 - chmod +x ./kubectl
-- sudo mv ./kubectl /usr/local/bin/kubectl
\ No newline at end of file
+- sudo mv ./kubectl /usr/local/bin/kubectl
+- chown -R "{{WrapAsParameter "jumpboxUsername"}}" "/home/{{WrapAsParameter "jumpboxUsername"}}"

```

Comment:
You could save a couple commands by using `chown` with user and group:

```
- chown -R "{{WrapAsParameter "jumpboxUsername"}}":"{{WrapAsParameter "jumpboxUsername"}}" "/home/{{WrapAsParameter "jumpboxUsername"}}"
- chown -R root:root "/home/{{WrapAsParameter "jumpboxUsername"}}/.kube"
```

Reply:
Except the fact, that not all implementations of chown support that, this depends on the tool chain. So my way is really portable.

Reply:
@mboersma  If I've not twisted my brain, the first two lines set the default of the home subtree owned by the user, and the last two set the ~/.kube subtree owned by root. That was the minimum change to satisfy the request, I'm not sore whether or not the user must be able to modify the config manually, he can nevertheless read it. This is a defensive approach, and it appears to work properly.